toolsrc/src/vcpkg/commands.format-manifest.cpp

```diff
@@ -172,20 +174,103 @@ Please open an issue at https://github.com/microsoft/vcpkg, with the following o
             }
         }
     }
+
+    static const auto CONTROL_name = fs::u8path("CONTROL");
+    static const auto vcpkg_json_name = fs::u8path("vcpkg.json");
+
+    Optional<ToWrite> read_input_file(Files::Filesystem& fs, const fs::path& resolved_path)
+    {
+        if (resolved_path.filename() == CONTROL_name)
+        {
+            return read_control_file(fs, resolved_path);
+        }
+
+        return read_manifest(fs, resolved_path);
+    }
+
+    void add_write(std::vector<std::string>& errors,
+                   std::vector<ToWrite>& to_write,
+                   Optional<ToWrite>&& this_manifest,
+                   const fs::path& resolved_path)
+    {
+        if (this_manifest.get())
+        {
+            to_write.push_back(std::move(this_manifest).value_or_exit(VCPKG_LINE_INFO));
+        }
+        else
+        {
+            errors.push_back(Strings::concat("Failed to parse ", fs::u8string(resolved_path)));
+        }
+    }
+
+    std::string format_both_manifest_and_control_error(StringView port_name)
+    {
+        return Strings::concat("Both a manifest file and a CONTROL file exist in port ", port_name, ".");
+    }
 }
 
 namespace vcpkg::Commands::FormatManifest
 {
+    ExpectedS<fs::path> resolve_format_manifest_input(StringView input,
+                                                      const fs::path& original_cwd,
+                                                      const fs::path& ports_base,
+                                                      const Files::ITestFileExists& filesystem)
+    {
+        auto p = fs::u8path(input);
+        if (p.is_absolute())
+        {
+            if (filesystem.exists(p, ignore_errors))
+            {
+                return std::move(p);
+            }
+
+            return {Strings::concat(input, " not found.")};
+        }
+
+        {
+            auto as_absolute = Files::combine(original_cwd, p);
+            if (filesystem.exists(as_absolute, ignore_errors))
+            {
+                return std::move(as_absolute);
+            }
+        } // destroy as_absolute
```

Comment:
```suggestion
        {
            auto p = Files::combine(original_cwd, fs::u8path(input));
            if (filesystem.exists(p, ignore_errors))
            {
                return std::move(p);
            }
        }
```

`Files::combine()` should do the right thing w.r.t. p being absolute. Additionally, I don't think the short-circuit here on `p.is_absolute() && !exists(p)` really makes a difference -- the code following still must handle relative paths that don't exist and it doesn't seem useful to carve out absolute paths here.